catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/query/suggestion/DmsCoordinateProcessorTest.java

```diff
@@ -0,0 +1,188 @@
+  public void testDmsStringAtLonBoundary() {
+    assertSuggestion(
+        "28°56\'26\"N 180°00\'00\"W",
+        "DMS: [ 28°56'26\"N 180°00'00\"W ]",
+        ImmutableList.of(new LatLon(28.940555, -180.0)));
+  }
+
+  @Test
+  public void testDmsStringAllZeros() {
+    assertSuggestion(
+        "0°0\'0\"S 0°0\'0\"E",
+        "DMS: [ 00°00'00\"S 000°00'00\"E ]",
+        ImmutableList.of(new LatLon(0.0, 0.0)));
+  }
+
+  @Test
+  public void testDmsStringNoDirection() {
+    assertSuggestionDoesNotExist("28°56\'26\" 117°38\'11\"");
+  }
+
+  @Test
+  public void testDmsStringLatDegreesTooLarge() {
+    assertSuggestionDoesNotExist("91°56\'26\"N 117°38\'11\"W");
+  }
+
+  @Test
+  public void testDmsStringLonDegreesTooLarge() {
+    assertSuggestionDoesNotExist("28°56\'26\"N 181°38\'11\"W");
+  }
+
+  @Test
+  public void testDmsStringLatMinutesTooLarge() {
+    assertSuggestionDoesNotExist("28°60\'26\"N 117°38\'11\"W");
+  }
+
+  @Test
+  public void testDmsStringLonMinutesTooLarge() {
+    assertSuggestionDoesNotExist("28°56\'26\"N 117°60\'11\"W");
+  }
+
+  @Test
+  public void testDmsStringLatSecondsTooLarge() {
+    assertSuggestionDoesNotExist("28°56\'60\"N 117°38\'11\"W");
+  }
+
+  @Test
+  public void testDmsStringLonSecondsTooLarge() {
+    assertSuggestionDoesNotExist("28°56\'26\"N 117°38\'60\"W");
+  }
+
+  @Test
+  public void testDmsStringConvertedLatTooLarge() {
+    assertSuggestionDoesNotExist("90°00\'01\"N 117°38\'11\"W");
+  }
+
+  @Test
+  public void testDmsStringConvertedLonTooLarge() {
+    assertSuggestionDoesNotExist("28°56\'26\"N 180°00\'01\"W");
+  }
+
+  @Test
+  public void testDmsStringMultipleCoordinates() {
+    assertSuggestion(
+        "28°56\'26\"N 117°38\'11.64564\"W 28;56;26S 117°38\'11\"QQW%WWEQW@!!!!!! 28°56\'26\"N 117°38\'11\"E",
+        "DMS: [ 28°56'26\"N 117°38'11.646\"W ] [ 28°56'26\"S 117°38'11\"W ] [ 28°56'26\"N 117°38'11\"E ]",
+        ImmutableList.of(
+            new LatLon(28.940555, -117.636568),
+            new LatLon(-28.940555, -117.636388),
+            new LatLon(28.940555, 117.636388)));
+  }
+
+  private void assertSuggestionDoesNotExist(String query) {
+    List<Suggestion> list = new LinkedList<>();
+    processor.enhanceResults(list, query);
+    assertThat(list, is(empty()));
+  }
+
+  private void assertSuggestion(String query, String expectedName, List<LatLon> expectedGeo) {
+    List<Suggestion> list = new LinkedList<>();
+    processor.enhanceResults(list, query);
+    assertThat(list, is(not(empty())));
+    assertThat(list, hasSize(1));
+
+    LiteralSuggestion literalSuggestion = (LiteralSuggestion) list.get(0);
+    assertThat(literalSuggestion.getId(), is("LITERAL-DMS"));
+    assertThat(literalSuggestion.getName(), is(expectedName));
+    assertThat(literalSuggestion.hasGeo(), is(true));
+
+    List<LatLon> actualGeo = literalSuggestion.getGeo();
+    List<LatLon> diff = new ArrayList<>();
+    assertThat(
+        "Actual geo and expected geo are the same size",
```

Comment:
This message is printed when the assertion fails so it should say that they are _not_ the same size.